### ISSUE TITLE
Docs: Text content does not match server-rendered HTML

### ISSUE THREAD

**Issue by little077** [vercel/next.js#67880]
### What is the update you wish to see?

But when I introduced some special JS packages using built-in component Script and didn't follow the documentation to use them（ https://nextjs.org/docs/messages/no-before-interactive-script-outside-document ）It also reports water injection errors. I spent an afternoon troubleshooting because the documentation for the former did not inform me that unexpected use of Script strategy="before Interactive" can also cause water injection issues. What's even more serious is when I changed the code in useMemo to something similar in the demo below const referLink = isLogin ?  "5" : "6"; At this point, the page won't report any errors, which is a strange phenomenon.

### Is there any context that might help us understand?

 This is my demo of reproducing it（ https://codesandbox.io/p/github/little077/demo/main ）

### Does the docs page already exist? Please link to it.

https://nextjs.org/docs/messages/react-hydration-error

**Comment by icyJoseph** [comment]
Hi,

The problem here is that, the script you are loading, looks for `link`-like text, and wraps it with a `<span>`, this is annoying because when hydration happens, it no longer finds just a text node, but a `<span>`, and that causes the error you are seeing. 

You can, maybe, try:

```jsx
        <div>
          卧槽了 <span suppressHydrationWarning>{referLink}</span>
        </div>
```

But that's not scalable, every single thing, that looks like a link, will get modified... and not to mention that, logic changes, will cause even more issues, similar to: https://martijnhols.nl/gists/everything-about-google-translate-crashing-react

**Comment by little077** [comment]
Are you saying that this package has done some operations on my code similar to sharing words? And during the operation, there happened to be water injection. Is my understanding correct.Also, since this can also cause water injection errors, should we also supplement it in the document

**Comment by icyJoseph** [comment]
Yeah, so, no issue here. Technically speaking though, `beforeInteractive` scripts, should be in `_document`:

> beforeInteractive scripts must be placed inside the Document Component (pages/_document.js) and are designed to load scripts that are needed by the entire site (i.e. the script will load when any page in the application has been loaded server-side).

- https://nextjs.org/docs/pages/api-reference/components/script#beforeinteractive

And, this duplicate issue, https://github.com/vercel/next.js/issues/67879... could you close both if you are satisfied with the answers?

**Comment by little077** [comment]
Okay, thank you very much. I saw the seventh common reason in the document. Maybe I didn't think of going here at first, it's really difficult to investigate, especially since my project has just been upgraded from 12 to 14. When I saw my reply, I had already closed this question
